core/src/test/scala/unit/kafka/metrics/MetricsTest.scala

```diff
@@ -166,9 +180,22 @@ class MetricsTest extends KafkaServerTestHarness with Logging {
       .count
   }
 
+  private def topicMetrics(topic: Option[String]): Set[String] = {
+    val metricNames = Metrics.defaultRegistry.allMetrics().keySet.asScala.map(_.getMBeanName)
+    filterByTopicMetricRegex(metricNames, topic)
+  }
+
   private def topicMetricGroups(topic: String): Set[String] = {
-    val topicMetricRegex = new Regex(".*BrokerTopicMetrics.*("+topic+")$")
     val metricGroups = Metrics.defaultRegistry.groupedMetrics(MetricPredicate.ALL).keySet.asScala
-    metricGroups.filter(topicMetricRegex.pattern.matcher(_).matches)
+    filterByTopicMetricRegex(metricGroups, Some(topic))
+  }
+
+  private def filterByTopicMetricRegex(metrics: Set[String], topic: Option[String]): Set[String] = {
+    val baseRegexPattern = ".*BrokerTopicMetrics.*"
```

Comment:
You can use `.r` on the string to create a regex. Also, instead of the pattern match, you can say something like:

```scala
baseRegexPattern + topic.map(t => s"($t)" + "$").getOrElse("")
```